extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -293,4 +694,72 @@ export class CssCalcProductNode extends CssNode {
     /** @const @private {string} */
     this.op_ = op;
   }
+
+  /** @override */
+  css() {
+    return `${this.left_.css()} ${this.op_} ${this.right_.css()}`;
+  }
+
+  /** @override */
+  isConst() {
+    return false;
+  }
+
+  /** @override */
+  calc(context) {
+    const left = this.left_.resolve(context);
+    const right = this.right_.resolve(context);
+    if (left == null || right == null) {
+      return null;
+    }
+    if (!(left instanceof CssNumericNode) ||
+        !(right instanceof CssNumericNode)) {
+      throw new Error('left and right must be both numerical');
+    }
+
+    /*
+     * From spec:
+     * At *, check that at least one side is <number>. If both sides are
+     * <integer>, resolve to <integer>. Otherwise, resolve to the type of the
+     * other side.
+     * At /, check that the right side is <number>. If the left side is
+     * <integer>, resolve to <number>. Otherwise, resolve to the type of the
+     * left side.
+     */
+    let base;
+    let multi;
+    if (this.op_ == '*') {
+      if (left instanceof CssNumberNode) {
+        multi = left.num_;
+        base = right;
+      } else {
+        if (!(right instanceof CssNumberNode)) {
+          throw new Error('one of sides in multiplication must be a number');
+        }
+        multi = right.num_;
+        base = left;
+      }
+    } else {
+      if (!(right instanceof CssNumberNode)) {
+        throw new Error('denominator must be a number');
+      }
+      base = left;
```

Comment:
I'm still investigating this part. But it appears that the right behavior is to resolve as `null`, as is done couple lines down. This, in most of cases, resolves to `none` in CSS.